ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java

```diff
@@ -435,4 +442,20 @@ public Instant getAddedToPoolAt() {
 
     TransactionSelectionResult evaluateTransaction(final Transaction transaction);
   }
+
+  public enum TransactionAddedStatus {
+    ALREADY_KNOWN(TransactionInvalidReason.TRANSACTION_ALREADY_KNOWN),
+    REJECTED_UNDERPRICED_REPLACEMENT(TransactionInvalidReason.TRANSACTION_REPLACEMENT_UNDERPRICED),
+    ADDED(null);
```

Comment:
I think it would be better to use optional rather than putting a null. What do you think ?

Reply:
Done.